src/haddock/clis/cli_pp.py

```diff
@@ -0,0 +1,53 @@
+"""HADDOCK3 PDB preprocessing client."""
+import argparse
+import sys
+
+from haddock.gear.preprocessing import process_pdbs, read_additional_residues
+from haddock.libs.libio import add_suffix_to_files, save_lines_to_files
+
+
+ap = argparse.ArgumentParser(
+    description=__doc__,
+    formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+
+ap.add_argument('pdb_files', help="Input PDB files.", nargs='+')
+ap.add_argument('-d', '--dry', help="Perform a dry run.", action="store_true")
+ap.add_argument('-t', '--topfile', help="Additional .top files.", nargs="*")
```

Comment:
This is nice, but for integrating this into other things (the servers), would be nice if there was an option that was not additive; `Exclusive .top files`? 

It could be that the list of supported residues is (becomes) different between services and then this pre-processing could only be used if whatever other application using it contains also supports all residues that are supported by haddock3 - which might not be the case.

Reply:
I will look carefully to make it as decoupled as possible from the allowed residues. But the implementation already is to some extent. Parameters can be given `None` to ignore them. And I can make it to pass a list of residues also, to avoid the `top` terminology.

Remember it is also possible to use the preprocessing from the library interface, which allows additional fine-tunning compared to the client.